index.js

```diff
@@ -84,6 +88,7 @@ function rateLimitPlugin (fastify, settings, next) {
           // load the global timewindow if it is missing from the route config
           routeOptions.config.rateLimit.timeWindow = mergedRateLimitParams.timeWindow
         }
+        routeOptions.config.rateLimit.timeWindow = formatTimeWindow(routeOptions.config.rateLimit.timeWindow)
```

Comment:
This is not necessary because the conversion from string to number has been done in the `makeParams`, check the `mergedRateLimitParams` var.

So we could just add:
```
.. else {
  routeOptions.config.rateLimit.timeWindow = mergedRateLimitParams.timeWindow
}
```

Reply:
You are right. So what do you think about this instead (remove that line and the entire if statement before it)
```
const mergedRateLimitParams = makeParams(routeOptions.config.rateLimit)
mergedRateLimitParams.routeInfo = routeOptions
current.store = pluginComponent.store.child(mergedRateLimitParams)
```

I think having ```mergedRateLimitParams``` look like the params that are passed to register the plugin is the intuitive approach but because we still need to do per route configuration, we pass in the ```routeInfo``` so we have method, url, path, etc... (including the entire config of the route). This also does guarantee the timeWindow provided as a number as you said but ```routeInfo.config.rateLimit.timeWindow``` is available in its original form just in case.

So like in RedisStore for example, that method then looks like:
```
RedisStore.prototype.child = function (routeOptions) {
  const child = Object.create(this)
  child.key = this.key + routeOptions.routeInfo.method + routeOptions.routeInfo.url + '-'
  child.timeWindow = routeOptions.timeWindow
  return child
}
```
No longer requires going into ```config.rateLimit.---``` but just expect the parameters at the top level of the object, just like the constructor does.

Thoughts?